controllers/contentlibrary/clustercontentlibraryitem/clustercontentlibraryitem_controller.go

```diff
@@ -202,6 +202,17 @@ func (r *Reconciler) setUpCVMIFromCCLItem(cvmi *vmopv1a1.ClusterVirtualMachineIm
 		return err
 	}
 
+	if cvmi.Labels == nil {
+		cvmi.Labels = make(map[string]string)
+	}
+
+	// Only watch for service type labels from ClusterContentLibraryItem
```

Comment:
You also  need to handle the case where the TKG library is changed  and the current TKG library is set as a cluster CL. In that case, the label  would be removed from the old CCLItems and hence should be removed from the CVMI resources as well.

Reply:
I think reconciliation will handle this situation. Once cclitem updates, controller will update the corresponding ClusterVirtualMachineImage resource.